Mathlib/CategoryTheory/Sites/RegularExtensive.lean

```diff
@@ -254,6 +256,33 @@ lemma isSheaf_of_projective (F : Cᵒᵖ ⥤ Type (max u v)) [Preregular C] [∀
   haveI : S.regular := ⟨Y, f, hh⟩
   exact isSheafFor_regular_of_projective _ _
 
+/-- Every Yoneda-presheaf is a sheaf for the regular topology. -/
+theorem isSheaf_yoneda_obj [Preregular C] (W : C)  :
+    Presieve.IsSheaf (regularCoverage C).toGrothendieck (yoneda.obj W) := by
+  rw [isSheaf_coverage]
+  intro X S ⟨_, hS⟩
+  have : S.regular := ⟨_, hS⟩
+  obtain ⟨Y, f, rfl, hf⟩ := Presieve.regular.single_epi (R := S)
+  have h_colim := isColimitOfEffectiveEpiStruct f hf.effectiveEpi.some
+  rw [← Sieve.generateSingleton_eq, ← Presieve.ofArrows_pUnit] at h_colim
+  intro x hx
+  let x_ext := Presieve.FamilyOfElements.sieveExtend x
+  have hx_ext := Presieve.FamilyOfElements.Compatible.sieveExtend hx
+  let S := Sieve.generate (Presieve.ofArrows (fun () ↦ Y) (fun () ↦ f))
+  obtain ⟨t, t_amalg, t_uniq⟩ :=
+    (Sieve.forallYonedaIsSheaf_iff_colimit S).mpr ⟨h_colim⟩ W x_ext hx_ext
+  refine ⟨t, ?_, ?_⟩
+  · convert Presieve.isAmalgamation_restrict (Sieve.le_generate
+      (Presieve.ofArrows (fun () ↦ Y) (fun () ↦ f))) _ _ t_amalg
+    exact (Presieve.restrict_extend hx).symm
+  · exact fun y hy ↦ t_uniq y <| Presieve.isAmalgamation_sieveExtend x y hy
+
+/-- The regular topology on any preregular category is subcanonical. -/
+theorem isSubcanonical [Preregular C] : Sheaf.Subcanonical (regularCoverage C).toGrothendieck :=
```

Comment:
This name would follow more `mathlib4` naming conventions.
```suggestion
theorem subcanonical [Preregular C] : Sheaf.Subcanonical (regularCoverage C).toGrothendieck :=
```
There is a similar fix which should be done at https://github.com/leanprover-community/mathlib4/blob/master/Mathlib/CategoryTheory/Sites/Coherent.lean#L132